storage/remote/codec.go

```diff
@@ -497,3 +497,24 @@ func metricTypeToMetricTypeProto(t textparse.MetricType) prompb.MetricMetadata_M
 
 	return prompb.MetricMetadata_MetricType(v)
 }
+
+// DecodeWriteRequest from a http.Request into a prompb.WriteRequest, handling
+// snappy decompression.
+func DecodeWriteRequest(r *http.Request) (*prompb.WriteRequest, error) {
```

Comment:
Should this take a Reader rather than an http request? That would allow more use cases in the future, and just make the caller pass r.Body in.

Reply:
Yeah, can do.  I was considering rolling a https://golang.org/pkg/net/http/#MaxBytesReader in there, but that will widen the inteface, not narrow it.  And that probably belongs in the handler which calls this.